modules/_labs/pagination/react/README.md

```diff
@@ -14,82 +74,661 @@ yarn add @workday/canvas-kit-labs-react-pagination
+the range.
+
 ```tsx
 import * as React from 'react';
-import Pagination from '@workday/canvas-kit-labs-react-pagination';
+import {
+  Pagination,
+  getLastPage,
+  getVisibleResultsMax,
+  getVisibleResultsMin,
+} from '../lib/Pagination';
+
+const PaginationWithGoToControls = () => {
+  const resultCount = 10;
+  const totalCount = 100;
+  const lastPage = getLastPage(resultCount, totalCount);
+
+  return (
+    <Pagination
+      aria-label="Pagination"
+      lastPage={lastPage}
+      rangeSize={3}
+      onPageChange={pageNumber => console.log(pageNumber)}
+    >
+      <Pagination.Controls>
+        <Pagination.JumpToFirstButton aria-label="First" />
+        <Pagination.StepToPreviousButton aria-label="Previous" />
+        <Pagination.PageList>
+          {({state}) =>
+            state.range.map(pageNumber => (
+              <Pagination.PageListItem key={pageNumber}>
+                <Pagination.PageButton aria-label={`Page ${pageNumber}`} pageNumber={pageNumber} />
+              </Pagination.PageListItem>
+            ))
+          }
+        </Pagination.PageList>
+        <Pagination.StepToNextButton aria-label="Next" />
+        <Pagination.JumpToLastButton aria-label="Last" />
+        <Pagination.GoToForm>
+          <Pagination.GoToTextInput />
+          <Pagination.GoToLabel>{() => `of ${totalCount} results`}</Pagination.GoToLabel>
+        </Pagination.GoToForm>
+      </Pagination.Controls>
+      <Pagination.AdditionalDetails shouldHideDetails>
+        {({state}) =>
+          `${getVisibleResultsMin(state.currentPage, resultCount)}-${getVisibleResultsMax(
+            state.currentPage,
+            resultCount,
+            totalCount
+          )} of ${totalCount} results`
+        }
+      </Pagination.AdditionalDetails>
+    </Pagination>
+  );
+};
+```
+
+---
+
+### Advanced Usage
+
+Below are examples for more advanced / custom usage of these components. We expect most people won't
+need to read this section, but if you're needing to go beyond our basic examples or just curious,
+feel free to explore this section. While these examples are not exhaustive, they provide additional
+insight into what's possible.
+
+#### Hoisted Model Pattern
+
+If you want the `Pagination` component to handle its state and actions internally and hook into page
+change events, the basic usage examples above should be sufficient. However, if you need external
+access to the model, you can use the hoisted model pattern. You can create a model outside of the
+component with the `usePaginationModel` hook.
+
+For this example, we'll create a `Pagination` component that handles 128 total items with 10 items
+per page. Normally this would involve some math on your part, but don't worry about that, we've got
+you covered with some handy utilities
+
+```tsx
+import {
+  getLastPage,
+  getVisibleResultsMax,
+  getVisibleResultsMin,
+} from '../lib/Pagination';
```

Comment:
```suggestion
import * as React from 'react';
import {
  Pagination,
  getLastPage,
  getVisibleResultsMax,
  getVisibleResultsMin,
  usePaginationModel,
} from '@workday/canvas-kit-labs-react-pagination';
```
It'll be a lot easier to catch these errors once these examples are actually rendered with Nicholas' docs improvements. :)